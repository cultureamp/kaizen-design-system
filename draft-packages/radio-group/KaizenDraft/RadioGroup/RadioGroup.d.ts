import * as React from "react";
export declare type RadioGroupProps = {
    automationId?: string;
    labelText: string | React.ReactNode;
    noBottomMargin?: boolean;
};
declare type RadioGroup = React.FunctionComponent<RadioGroupProps>;
declare const RadioGroup: RadioGroup;
export default RadioGroup;
