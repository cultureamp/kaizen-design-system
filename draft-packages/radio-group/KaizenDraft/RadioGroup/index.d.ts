export { default as RadioGroup } from "./RadioGroup";
