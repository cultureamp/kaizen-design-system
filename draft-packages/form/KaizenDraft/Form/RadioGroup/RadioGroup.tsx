import React, { useState, HTMLAttributes } from "react"
import classnames from "classnames"
import { v4 } from "uuid"
import { OverrideClassName } from "@kaizen/component-base"
import { Label } from "../Primitives"
import styles from "./RadioGroup.module.scss"

export interface RadioGroupProps
  extends OverrideClassName<HTMLAttributes<HTMLDivElement>> {
  children?: React.ReactNode
  labelText: string | React.ReactNode
  labelId?: string
  noBottomMargin?: boolean
  reversed?: boolean
  /**
   * **Deprecated:** Use test id compatible with your testing library (eg. `data-testid`).
   * @deprecated
   */
  automationId?: string
}

export const RadioGroup = ({
  children,
  labelId: propsLabelId,
  labelText,
  noBottomMargin = false,
  reversed = false,
  automationId = "",
  classNameOverride,
  ...restProps
}: RadioGroupProps): JSX.Element => {
  const [labelId] = useState<string>(propsLabelId || v4())
  return (
    <div
      data-automation-id={automationId}
      className={classnames(
        styles.radioGroupContainer,
        classNameOverride,
        noBottomMargin && styles.noBottomMargin,
        reversed && styles.reversed
      )}
      role="radiogroup"
      aria-labelledby={labelId}
      {...restProps}
    >
      <div className={styles.radioGroupLabel}>
        <Label
          id={labelId}
          automationId={automationId && `${automationId}-field-label`}
          labelText={labelText}
          labelType="text"
          reversed={reversed}
        />
      </div>
      {children}
    </div>
  )
}

RadioGroup.displayName = "RadioGroup"
