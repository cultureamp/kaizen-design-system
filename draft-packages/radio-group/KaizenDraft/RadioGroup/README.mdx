---
title: Radio Group
navTitle: Radio Group
summaryParagraph: Radio groups include a set of radios that let people select exactly one option in a set.
tags: ["Choice list", "Radio button", "Radio button group", "Option button", "Choice", "Selection"]
needToKnow:
- Prefer vertical (stacked) radios to easily scan and compare options.
- When there’s only two options, consider if a single checkbox would be better.
- For settings that are immediately applied, consider using a toggle switch instead.
demoStoryId: radiogroup-react--default-kaizen-site-demo
headerImage: radio-group
---

import WhenToUseAndWhenNotToUse from "docs-components/WhenToUseAndWhenNotToUse"
import WhenToUse from "docs-components/WhenToUse"
import WhenNotToUse from "docs-components/WhenNotToUse"

import DoAndDontContainer from "docs-components/DoAndDontContainer"
import Do from "docs-components/Do"
import Dont from "docs-components/Dont"

## Visuals

### UI Kit

<iframe style="border: none;" width="744" height="450" src="https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Ffile%2Fe3TyAOZKfLg9cA01Iy1Mgm%2FKaizen-Site-embed%3Fnode-id%3D1412%253A270" allowfullscreen></iframe>

<iframe style="border: none;" width="744" height="450" src="https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Ffile%2Fe3TyAOZKfLg9cA01Iy1Mgm%2FKaizen-Site-embed%3Fnode-id%3D1412%253A277" allowfullscreen></iframe>

## Copy guidelines

Radios are helpful when presenting a set of mutually exclusive options out of which a user must make one selection. Keep these points in mind:

### Present mutually exclusive options

Each option in a radio group should be mutually exclusive from the other to make it easier for a user to make a selection.

<DoAndDontContainer>
<Do>

- Select your age group:
    - 18-24
    - 25-34
    - 35-44
    - 45-54
    - 55+

</Do>

<Dont>

- Your age group is...
    - 18-24,
    - 24-34,
    - 34-44,
    - 44-54,
    - 54+

</Dont>
</DoAndDontContainer>

### Always introduce options with a label
Always present radio options with an intro sentence (or label) that describes the choice the user has to make. End the label with a colon.

### Don’t end options in punctuation
If your options are single sentences, words or fragments, don’t place a period, comma or semicolon at the end of them. Always err on the side of minimal punctuation, so leave out full stops and commas where possible.

### Write in sentence case
All radio copy should be in sentence case i.e. the first letter of the first word is capitalized and everything else is in lower case unless it’s a proper noun or feature name.

## To keep in mind

*   There must always be more than one radio field used together in a group.
*   Radio Group contains a set of Radios and a Radio Group Label.
*   Show options in a sensible order, such as alphabetical, numeric.
*   Avoid long lists. Chunk long lists and group related options together.
*   The setting only applies when the user takes an action, such as clicking a save button.
*   Prefer vertical (stacked) radios to easily scan and compare options.
*   Use a fieldset to group related radio fields together in a set.
*   Offer a default option.
*   Prefer to use a default selected option.
*   Avoid nesting.
*   For Right-To-Left languages, add `dir=”rtl”` to a `div` wrapping the Radios
*   Disabled state:
    *   See [Interaction states guidelines for disabled elements](/guidelines/interaction-states#disabled-state).

## When to use and when not to use

<WhenToUseAndWhenNotToUse>
<WhenToUse>

*   Single choice options.
*   Shown side-by-side or stacked.
*   Where there is space to show them all at once.

</WhenToUse>
<WhenNotToUse>

*   Don’t use when there is more than 20 options.
*   When there’s only two options, consider if a single [checkbox](/components/checkbox-field) would be better.
*   For settings that are immediately applied, consider using a [toggle switch](/components/toggle-switch-field) instead.
*   When there’s limited space, consider using a select dropdown instead.
*   Sometimes, a dropdown menu might be better. If it’s an infrequently used setting, has a good default, has 4 or more options, consider using a select dropdown menu.

</WhenNotToUse>
</WhenToUseAndWhenNotToUse>

## See also

Some websites have a “radio button group” concept where radio options are styled like buttons and you can only select one in the group. That’s a different thing to this.

## External Links

Here are some examples of other existing design systems:

- [Material: radio button](https://material.io/develop/web/components/input-controls/radio-buttons/)
- [Material: form fields](https://material.io/develop/web/components/input-controls/form-fields/)
- [Lightning: radio group](https://www.lightningdesignsystem.com/components/radio-group/)
- [Polaris: choice list](https://polaris.shopify.com/components/forms/choice-list)
- [Polaris: radio button](https://polaris.shopify.com/components/forms/radio-button)
- Copy resource: [Nielsen Norman Group: Checkboxes vs. radio buttons](https://www.nngroup.com/articles/checkboxes-vs-radio-buttons/)
- Copy resource: [UX Planet: Radio buttons](https://uxplanet.org/radio-buttons-ux-design-588e5c0a50dc)
