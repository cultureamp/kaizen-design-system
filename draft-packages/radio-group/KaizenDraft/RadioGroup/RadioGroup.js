"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
exports.__esModule = true;
var classnames_1 = __importDefault(require("classnames"));
var React = __importStar(require("react"));
var draft_1 = require("@kaizen/component-library/draft");
var styles = require("./styles.scss");
var RadioGroup = function (_a) {
    var _b;
    var _c = _a.automationId, automationId = _c === void 0 ? "" : _c, children = _a.children, _d = _a.labelText, labelText = _d === void 0 ? "" : _d, _e = _a.noBottomMargin, noBottomMargin = _e === void 0 ? false : _e;
    return (React.createElement("div", { "data-automation-id": automationId, className: classnames_1["default"](styles.radioGroupContainer, (_b = {},
            _b[styles.noBottomMargin] = noBottomMargin,
            _b)) },
        React.createElement("div", { className: styles.radioGroupLabel },
            React.createElement(draft_1.Label, { automationId: automationId + "-field-label", labelText: labelText, labelType: "text" })),
        children));
};
exports["default"] = RadioGroup;
//# sourceMappingURL=RadioGroup.js.map