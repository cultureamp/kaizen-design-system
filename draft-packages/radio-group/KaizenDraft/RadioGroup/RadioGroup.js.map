{"version":3,"file":"RadioGroup.js","sourceRoot":"","sources":["RadioGroup.tsx"],"names":[],"mappings":";;;;;;;;;;;;AAAA,0DAAmC;AACnC,2CAA8B;AAE9B,yDAAuD;AAEvD,IAAM,MAAM,GAAG,OAAO,CAAC,eAAe,CAAC,CAAA;AAUvC,IAAM,UAAU,GAAe,UAAC,EAK/B;;QAJC,oBAAiB,EAAjB,sCAAiB,EACjB,sBAAQ,EACR,iBAAc,EAAd,mCAAc,EACd,sBAAsB,EAAtB,2CAAsB;IAClB,OAAA,CACJ,mDACsB,YAAY,EAChC,SAAS,EAAE,uBAAU,CAAC,MAAM,CAAC,mBAAmB;YAC9C,GAAC,MAAM,CAAC,cAAc,IAAG,cAAc;gBACvC;QAEF,6BAAK,SAAS,EAAE,MAAM,CAAC,eAAe;YACpC,oBAAC,aAAK,IACJ,YAAY,EAAK,YAAY,iBAAc,EAC3C,SAAS,EAAE,SAAS,EACpB,SAAS,EAAC,MAAM,GAChB,CACE;QACL,QAAQ,CACL,CACP;AAhBK,CAgBL,CAAA;AAED,qBAAe,UAAU,CAAA"}