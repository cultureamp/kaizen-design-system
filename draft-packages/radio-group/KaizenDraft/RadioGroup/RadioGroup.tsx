import classnames from "classnames"
import * as React from "react"

import { Label } from "@kaizen/component-library/draft"

const styles = require("./styles.scss")

export type RadioGroupProps = {
  automationId?: string
  labelText: string | React.ReactNode
  noBottomMargin?: boolean
}

type RadioGroup = React.FunctionComponent<RadioGroupProps>

const RadioGroup: RadioGroup = ({
  automationId = "",
  children,
  labelText = "",
  noBottomMargin = false,
}) => (
  <div
    data-automation-id={automationId}
    className={classnames(styles.radioGroupContainer, {
      [styles.noBottomMargin]: noBottomMargin,
    })}
  >
    <div className={styles.radioGroupLabel}>
      <Label
        automationId={`${automationId}-field-label`}
        labelText={labelText}
        labelType="text"
      />
    </div>
    {children}
  </div>
)

export default RadioGroup
