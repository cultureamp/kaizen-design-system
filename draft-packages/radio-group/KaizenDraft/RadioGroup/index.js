"use strict";
exports.__esModule = true;
var RadioGroup_1 = require("./RadioGroup");
exports.RadioGroup = RadioGroup_1["default"];
//# sourceMappingURL=index.js.map