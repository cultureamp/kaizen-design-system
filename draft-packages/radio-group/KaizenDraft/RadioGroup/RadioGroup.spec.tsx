import { Radio } from "@kaizen/component-library/draft"
import { cleanup, fireEvent, render } from "@testing-library/react"
import * as React from "react"
import RadioGroup, { RadioGroupProps } from "./RadioGroup"

afterEach(cleanup)

const defaultRadioGroupProps: RadioGroupProps = {
  automationId: "RadioAutomationId",
  labelText: "Label",
}
const renderRadioGroup = (props?: RadioGroupProps) => {
  const mergedRadioGroupProps = { ...defaultRadioGroupProps, ...props }

  return render(<RadioGroup {...mergedRadioGroupProps} />)
}

describe("<RadioGroup /> ", () => {
  describe("snapshots", () => {
    test("renders RadioGroup in a column and with a label", () => {
      const { container } = render(
        <RadioGroup labelText="Label" children={null} />
      )

      expect(container.firstChild).toMatchSnapshot()
    })

    test("renders RadioGroup in a row and with a label", () => {
      const { container } = render(
        <RadioGroup labelText="Label" children={null} />
      )

      expect(container.firstChild).toMatchSnapshot()
    })
    test("renders RadioGroup with radios", () => {
      const { container } = render(
        <RadioGroup labelText="Label">
          <Radio
            name="radio"
            id="radio-1"
            labelText="Label 1"
            selectedStatus={false}
            disabled={false}
            value="radio-1"
          />
          <Radio
            name="radio"
            id="radio-2"
            labelText="Label 2"
            selectedStatus={false}
            disabled={false}
            value="radio-2"
          />
        </RadioGroup>
      )

      expect(container.firstChild).toMatchSnapshot()
    })

    test("renders RadioGroup in a column and with a label", () => {
      const { container } = render(
        <RadioGroup labelText="Label" children={null} />
      )

      expect(container.firstChild).toMatchSnapshot()
    })
  })
  describe("presentational", () => {
    it("should render a title", () => {
      const title = "Label"

      const { queryByText } = render(
        <RadioGroup labelText={title} children={null} />
      )
      expect(queryByText(title)).toBeTruthy()
    })
  })
})
