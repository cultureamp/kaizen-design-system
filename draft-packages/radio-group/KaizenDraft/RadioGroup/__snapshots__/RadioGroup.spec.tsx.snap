// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup />  snapshots renders RadioGroup in a column and with a label 1`] = `
<div
  class="radioGroupContainer"
  data-automation-id=""
>
  <div
    class="radioGroupLabel"
  >
    <label
      class="label text"
      data-automation-id="-field-label"
    >
      <span
        class="prependedLabel"
      >
        Label
      </span>
    </label>
  </div>
</div>
`;

exports[`<RadioGroup />  snapshots renders RadioGroup in a column and with a label 2`] = `
<div
  class="radioGroupContainer"
  data-automation-id=""
>
  <div
    class="radioGroupLabel"
  >
    <label
      class="label text"
      data-automation-id="-field-label"
    >
      <span
        class="prependedLabel"
      >
        Label
      </span>
    </label>
  </div>
</div>
`;

exports[`<RadioGroup />  snapshots renders RadioGroup in a row and with a label 1`] = `
<div
  class="radioGroupContainer"
  data-automation-id=""
>
  <div
    class="radioGroupLabel"
  >
    <label
      class="label text"
      data-automation-id="-field-label"
    >
      <span
        class="prependedLabel"
      >
        Label
      </span>
    </label>
  </div>
</div>
`;

exports[`<RadioGroup />  snapshots renders RadioGroup with radios 1`] = `
<div
  class="radioGroupContainer"
  data-automation-id=""
>
  <div
    class="radioGroupLabel"
  >
    <label
      class="label text"
      data-automation-id="-field-label"
    >
      <span
        class="prependedLabel"
      >
        Label
      </span>
    </label>
  </div>
  <div
    class="container"
  >
    <label
      class="label radio"
      data-automation-id="radio-1-field-label"
      for="radio-1"
      id="radio-1-field-label"
    >
      <div>
        <input
          class="radioInput needsclick"
          data-automation-id="radio-1-radio-input"
          id="radio-1"
          name="radio"
          type="radio"
          value="radio-1"
        />
        <div
          class="box"
        />
      </div>
      <span
        class="prependedLabel"
      >
        Label 1
      </span>
    </label>
  </div>
  <div
    class="container"
  >
    <label
      class="label radio"
      data-automation-id="radio-2-field-label"
      for="radio-2"
      id="radio-2-field-label"
    >
      <div>
        <input
          class="radioInput needsclick"
          data-automation-id="radio-2-radio-input"
          id="radio-2"
          name="radio"
          type="radio"
          value="radio-2"
        />
        <div
          class="box"
        />
      </div>
      <span
        class="prependedLabel"
      >
        Label 2
      </span>
    </label>
  </div>
</div>
`;
